tests/unit/trainer/test_trainer.py

```diff
@@ -59,6 +64,45 @@ def trainer(float_tolerance):
         yield c
 
 
+@pytest.fixture(scope="function")
+def trainer_w_percent_n_train_n_val(float_tolerance):
+    """
+    Generate a class instance with minimal configurations,
+    where n_train and n_val are given as percentage of the
+    dataset size.
+    """
+    conf = minimal_config.copy()
+    conf["n_train"] = N_TRAIN_PERCENT
+    conf["n_val"] = N_VAL_PERCENT  # note that summed percentages don't have to be 100%
+    conf["default_dtype"] = str(torch.get_default_dtype())[len("torch.") :]
+    model = model_from_config(conf)
+    with tempfile.TemporaryDirectory(prefix="output") as path:
+        conf["root"] = path
+        c = Trainer(model=model, **conf)
+        yield c
+
+
+@pytest.fixture(scope="function")
+def trainer_w_percent_n_train_n_val_flooring(float_tolerance):
+    """
+    Generate a class instance with minimal configurations,
+    where n_train and n_val are given as percentage of the
+    dataset size, summing to 100% but with a split that gives
+    non-integer numbers of frames for n_train and n_val.
+    (i.e. n_train = 70% = 5.6 frames, n_val = 30% = 2.4 frames,
+    so final n_train is 6 and n_val is 2)
+    """
+    conf = minimal_config.copy()
+    conf["n_train"] = N_TRAIN_PERCENT_100
+    conf["n_val"] = N_VAL_PERCENT_100
+    conf["default_dtype"] = str(torch.get_default_dtype())[len("torch.") :]
+    model = model_from_config(conf)
+    with tempfile.TemporaryDirectory(prefix="output") as path:
```

Comment:
It doesn't matter much, but conceptually I think this test setup belongs inside the tests below, since these objects are never reused, and having the `N_TRAIN_*` and `N_VAL_*` variables as globals is a little unusual.

Reply:
Good points, this is now solved with [27dcae0](https://github.com/mir-group/nequip/pull/438/commits/27dcae02340d9728c34edf0947604cea0d56df71)
Avoids duplicating the `trainer` generation and instead allows generation of slightly modified `Trainer` using kwargs, now used in the corresponding tests, as well as using `pytest.mark.parametrize` to test a few more train/val splits. 👍